pac4j-core/src/main/java/org/pac4j/core/profile/Pac4JPrincipal.java

```diff
@@ -0,0 +1,61 @@
+package org.pac4j.core.profile;
+
+import java.security.Principal;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+
+import org.pac4j.core.util.CommonHelper;
+
+public class Pac4JPrincipal implements Principal {
+
+    private final CommonProfile profile;
+
+    private final Set<String> roles;
+
+    public Pac4JPrincipal(LinkedHashMap<String, CommonProfile> profiles) {
+        Optional<CommonProfile> optProfile = ProfileHelper.flatIntoOneProfile(profiles);
+        if (!optProfile.isPresent()) {
+            throw new IllegalArgumentException();
```

Comment:
Let's throw a `TechnicalException` with a clear message to stay in the "pac4j" field...